packages/builder/builder-shared/src/apply/output.ts

```diff
@@ -82,7 +82,13 @@ function getPublicPath({
   } else if (dev.assetPrefix === true) {
     const hostname = context.devServer?.hostname || DEFAULT_DEV_HOST;
     const port = context.devServer?.port || DEFAULT_PORT;
-    publicPath = `//${hostname}:${port}/`;
+    if (hostname !== DEFAULT_DEV_HOST) {
```

Comment:
Maybe we should revert this Pull Request: https://github.com/web-infra-dev/modern.js/pull/2975/files

0.0.0.0 should not be used as default dev host, it is not same as `localhost`:

> In the context of a Windows system, 0.0.0.0 and localhost are not equivalent.

> 0.0.0.0 is a special IP address that represents all available IP addresses on the local machine. When a server is configured to bind to 0.0.0.0, it will listen on all available network interfaces, including the loopback interface (127.0.0.1) and any physical network interfaces.

> localhost, on the other hand, is a hostname that resolves to the loopback IP address (127.0.0.1). It is commonly used to refer to the local machine and is often used as a default hostname for local web servers.

> So while both 0.0.0.0 and localhost can be used to refer to the local machine, they have different meanings and uses.